### PR TITLE
feat(unlock-app) - Verifiers page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

This PR is part of https://github.com/unlock-protocol/unlock/issues/8663, and add verifier page

### verifier list page
verifiers-list-page

### add verifier modal
add-verifier

### delete verifier modal
delete-verifier

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Hum, I think you broke the template :D

**Comment by julien51** [comment]
Please, make sure you add screenshots...

File: unlock-app/src/components/content/VerifiersContent.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          headers: { 'Content-Type': 'application/json' },
+        }
+        await fetch(addVerifierUrl, requestOptions)
+          .then((res) => res.json())
+          .then(() => {
+            ToastHelper.success('Verifier added to list')
+          })
+        setLoading(false)
+      } else {
+        ToastHelper.error('Recipient is not valid, please check it again')
```
Comment (julien51) [inline unlock-app/src/components/content/VerifiersContent.tsx:63]:
```suggestion
        ToastHelper.error('Verified address is not a valid Ethereum address.')
```

File: unlock-app/src/components/content/VerifiersContent.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        setLoading(false)
+      }
+    } catch (err: any) {
+      setLoading(false)
+      ToastHelper.error(
+        err?.error ??
+          'There was a problem adding verifier, please re-load and try again'
```
Comment (julien51) [inline unlock-app/src/components/content/VerifiersContent.tsx:70]:
```suggestion
          'There was a problem adding the verifier address, please re-load and try again'
```

File: unlock-app/src/components/content/VerifiersContent.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          <Button onClick={onAddVerifier}>Add verifier</Button>
+        </Header>
+        <VerifiersList lockAddress={lockAddress} />
+      </VerifierContent>
+
+      <Modal isOpen={addVerifierModalOpen} setIsOpen={onAddVerifier}>
+        <div className={styling.sectionWrapper}>
+          <h3 className={styling.sectionTitle}>Add verifier</h3>
+          <span className={styling.sectionDesctiption}>
+            Input the wallet address of the user you wamt to give permission
```
Comment (julien51) [inline unlock-app/src/components/content/VerifiersContent.tsx:93]:
```suggestion
            Enter the Ethereum address of the user you want to add as a Verifier.
```

File: unlock-app/src/components/interface/verifiers/VerifiersList.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      await fetch(addVerifierUrl, requestOptions)
+        .then((res) => res.json())
+        .then((verifiers: any) => {
+          setVerifiers(verifiers)
+        })
+    } catch (err: any) {
+      setLoading(false)
+      ToastHelper.error(
+        err?.error ??
+          'There was a problem adding verifier, please re-load and try again'
```
Comment (julien51) [inline unlock-app/src/components/interface/verifiers/VerifiersList.tsx:57]:
```suggestion
          'We could not load the list of verifiers for your lock. Please reload to to try again.'
```

File: unlock-app/src/components/interface/verifiers/VerifiersList.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        await fetch(addVerifierUrl, requestOptions)
+          .then((res) => res.json())
+          .then(() => {
+            ToastHelper.success('Verifier deleted from list')
+          })
+        setDefaults()
+        setTimeout(() => {
+          // reload page to get updated list after delete
+          getVerifierList()
+        }, 2000)
```
Comment (julien51) [inline unlock-app/src/components/interface/verifiers/VerifiersList.tsx:80]: Do we need that?

File: unlock-app/src/components/interface/verifiers/VerifiersList.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      setLoading(false)
+      ToastHelper.error(
+        err?.error ??
+          'There was a problem adding verifier, please re-load and try again'
+      )
+    }
+  }
+  const onConfirmDeleteVerifier = async () => {
+    setLoading(true)
+    const resolvedAddress = await getAddressForName(selectedVerifier)
```
Comment (julien51) [inline unlock-app/src/components/interface/verifiers/VerifiersList.tsx:63]: Can we use the address directly and not the ENS here? I assume we store addresses and not ENS on our end.

Comment (kalidiagne) [reply unlock-app/src/components/interface/verifiers/VerifiersList.tsx:63]: ok! nice

**Review by julien51 — APPROVED** [review]

**Pull request merged**